portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -15,11 +20,32 @@
 public class DeleteCommentServlet extends HttpServlet {
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      long id = Long.parseLong(request.getParameter("id"));
+      Key commentEntityKey = KeyFactory.createKey("Comment", id);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-    Key commentEntityKey = KeyFactory.createKey("Comment", id);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(commentEntityKey);
+      // get email of the user who wrote the comment
+      Entity commentEntity = datastore.get(commentEntityKey);
+      String authorEmail = (String) commentEntity.getProperty("email");
+
+      // get email of current user who is trying to delete comment
+      UserService userService = UserServiceFactory.getUserService();
+      String currentEmail = userService.getCurrentUser().getEmail();
+
+      response.setContentType("text/html");
+      PrintWriter out = response.getWriter();
+
+      if (currentEmail.equals(authorEmail)) {
+        datastore.delete(commentEntityKey);
+        out.append("Allowed");
+      } else {
+        out.append("Not allowed");
+      }
+      out.close();
+    } catch (EntityNotFoundException e) {
+      System.out.println("Can't fetch entity");
+    }
```

Comment:
I think rather than returning a string in the response text, you could use the response status_code. Permission_denied seems appropriate here